nss-day-cohort-29/reactive-nutshell-springwynne
# Ac user login

# Description
Users can now login and register new accounts. Session storage is set at register. Fixed small bug in tasks

## Type of change

- [ X ] Bug fix (non-breaking change which fixes an issue)
- [ X ] New feature (non-breaking change which adds functionality)


# Testing Instructions for Change Made

1. git checkout `ac-user-login`
1. make sure json server is running
1. navigate to localhost:3000/login
1. attempt to login as a user that does not exist. Make sure that alert pops up that says "We can't seem to find you! Try registering below"
1. click on "Register Here" link. Make sure that it "re-directs" you to http://localhost:3000/register
1. enter a username that already exists, make sure that an alert pops up that says "Sorry, this username is taken!"
1. Enter a unique UserName, password and email. When you press register, make sure it redirects you  to http://localhost:3000/ and that the UserName, password and email are added to the database.json under "users"
1. navigate back to http://localhost:3000/login and login with the newly registered information, make sure you are redirected to http://localhost:3000/

# Checklist:

- [ X ] My code follows the style guidelines of this project
- [ X ] I have performed a self-review of my own code
- [ X ] My changes generate no new warnings
- [ X ] I have added test instructions that prove my fix is effective or that my feature works`
